docs/docs/reference/release-notes/migrating-source-plugin-from-v3-to-v4.md

```diff
@@ -0,0 +1,276 @@
+# Upgrade your source plugins for Gatsby 4
+
+Gatsby 4 is here! Following on the heels of Gatsby 3, Gatsby 4 further improves build performance and introduces new parallel processing capabilities. In the guide below, we'll walk you through preparing your source plugin for Gatsby 4.
+
+Introducing support for Gatsby 4 in your source plugin is straightforward and can be accomplished in just a few days. With Gatsby 4, Core APIs are being split into different processes so they're able to run simultaneously in parallel instead of sequentially. Early results show at least 40% improvement of build performance.
+
+Gatsby 4 also lays the groundwork for two new methods of rendering your Gatsby site: Server Side Rendering (SSR) and Deferred Static Rendering (DSR) to scale Gatsby to infinity and beyond.
+
+Let's get into it! In the guide below, we'll outline the many breaking changes in Gatsby 4 and some quick ways we've found to resolve these. Checkout `gatsby-source-wordpress` and `gatsby-source-shopify` to see how we introduced support for Gatsby 4.
```

Comment:
Wordpress had multiple PRs, Shopify had one but it only handled one aspect of the breaking changes, the rest of the plugin already abided by what we needed. So I think the idea was to just check out the plugin and see how the code can be organized, rather than how to move the code around (we show that below)

Reply:
Cool, in that case I think it makes sense to just link to the plugin page for each of those. I'll add a commit for it!